yarn-project/telemetry-client/src/lmdb_metrics.ts

```diff
@@ -1,38 +1,47 @@
-import { type Gauge, type Meter, type Metrics, ValueType } from './telemetry.js';
+import { BatchObservableResult, type Meter, type Metrics, ObservableGauge, ValueType } from './telemetry.js';
 
 export type LmdbMetricDescriptor = {
   name: Metrics;
   description: string;
 };
 
+export type LmdbStatsCallback = () => { mappingSize: number; numItems: number; actualSize: number };
+
 export class LmdbMetrics {
-  private dbMapSize: Gauge;
-  private dbUsedSize: Gauge;
-  private dbNumItems: Gauge;
+  private dbMapSize: ObservableGauge;
+  private dbUsedSize: ObservableGauge;
+  private dbNumItems: ObservableGauge;
 
   constructor(
     meter: Meter,
     dbMapSizeDescriptor: LmdbMetricDescriptor,
     dbUsedSizeDescriptor: LmdbMetricDescriptor,
     dbNumItemsDescriptor: LmdbMetricDescriptor,
+    private getStats?: LmdbStatsCallback,
   ) {
-    this.dbMapSize = meter.createGauge(dbMapSizeDescriptor.name, {
+    this.dbMapSize = meter.createObservableGauge(dbMapSizeDescriptor.name, {
       description: dbMapSizeDescriptor.description,
       valueType: ValueType.INT,
     });
-    this.dbUsedSize = meter.createGauge(dbUsedSizeDescriptor.name, {
+    this.dbUsedSize = meter.createObservableGauge(dbUsedSizeDescriptor.name, {
       description: dbUsedSizeDescriptor.description,
       valueType: ValueType.INT,
     });
-    this.dbNumItems = meter.createGauge(dbNumItemsDescriptor.name, {
+    this.dbNumItems = meter.createObservableGauge(dbNumItemsDescriptor.name, {
       description: dbNumItemsDescriptor.description,
       valueType: ValueType.INT,
     });
-  }
 
-  public recordDBMetrics(metrics: { mappingSize: number; numItems: number; actualSize: number }) {
-    this.dbMapSize.record(metrics.mappingSize);
-    this.dbNumItems.record(metrics.actualSize);
-    this.dbUsedSize.record(metrics.actualSize);
+    meter.addBatchObservableCallback(this.recordDBMetrics, [this.dbMapSize, this.dbUsedSize, this.dbNumItems]);
   }
+
+  private recordDBMetrics = (observable: BatchObservableResult) => {
+    if (!this.getStats) {
+      return;
+    }
+    const metrics = this.getStats();
+    observable.observe(this.dbMapSize, metrics.mappingSize);
+    observable.observe(this.dbNumItems, metrics.actualSize);
```

Comment:
Oh this is a typo on my part. That should be `metrics.numItems`.

Reply:
Fixed! Totally missed it haha, nice catch